flagsmith-core.ts

```diff
@@ -250,7 +262,12 @@ const Flagsmith = class {
                         resolve();
                     }
                 }).catch(({ message }) => {
-                    this.isLoading = false;
+                    this.setLoadingState({
```

Comment:
Instead of doing this in multiple places, I think we should just edit onError


```
       this.onError = onError? (message:any)=> {
                if (message instanceof Error) {
                    onError(message)
                } else {
                    onError(new Error(message))
                }
            }: null
```

To
```
this.onError = ()=> {
                    this.setLoadingState({...})

                 if (message instanceof Error) {
                    onError(message)
                } else {
                    onError(new Error(message))
                }
}
```